main.tf

```
@@ -0,0 +1,450 @@
+  name = "${var.app_name}-disaster-recovery-lambda-policy-${var.environment}"
+  role = aws_iam_role.disaster_recovery_lambda_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents",
+          "logs:DescribeLogGroups",
+          "logs:DescribeLogStreams"
+        ]
+        Resource = "arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:*"
+      },
+      # Table-specific DynamoDB permissions
+      {
+        Effect = "Allow"
+        Action = [
+          "dynamodb:CreateTable",
+          "dynamodb:DescribeTable",
+          "dynamodb:PutItem",
+          "dynamodb:BatchWriteItem",
+          "dynamodb:UpdateTable",
+          "dynamodb:DeleteTable",
+          "dynamodb:ListTables",
+          "dynamodb:DescribeContinuousBackups",
+          "dynamodb:UpdateContinuousBackups",
+          "dynamodb:ListTagsOfResource",
+          "dynamodb:TagResource",
+          "dynamodb:UntagResource",
+          "dynamodb:Scan",
+          "dynamodb:DeleteItem",
+          "dynamodb:GetItem",
+          "dynamodb:Query",
+          "dynamodb:UpdateItem"
+        ]
+        Resource = concat(
+          [for table_name in local.table_names :
+            "arn:aws:dynamodb:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:table/${table_name}"
+          ],
+          [for table_name in local.table_names :
+            "arn:aws:dynamodb:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:table/${table_name}/index/*"
+          ]
+        )
+      },
+      # DynamoDB Import Operations (requires Resource = "*")
+      {
+        Effect = "Allow"
+        Action = [
+          "dynamodb:ImportTable",
+          "dynamodb:DescribeImport",
+          "dynamodb:ListImports"
+        ]
+        Resource = "*"
+      },
+      # Global Table Operations
+      {
+        Effect = "Allow"
+        Action = [
+          "dynamodb:DescribeGlobalTable",
+          "dynamodb:CreateGlobalTable",
+          "dynamodb:UpdateGlobalTable"
+        ]
+        Resource = "*"
+      },
+
+      # S3 permissions for reading backups
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:GetObject",
+          "s3:ListBucket",
+          "s3:GetObjectVersion",
+          "s3:ListBucketVersions"
+        ]
+        Resource = [
+          data.aws_s3_bucket.mfa_backups.arn,
+          "${data.aws_s3_bucket.mfa_backups.arn}/*"
+        ]
+      },
+      # CloudWatch for monitoring import progress
+      {
+        Effect = "Allow"
+        Action = [
+          "cloudwatch:PutMetricData"
+        ]
+        Resource = "*"
+        Condition = {
+          StringEquals = {
+            "cloudwatch:namespace" = "${var.app_name}/DisasterRecovery"
+          }
+        }
+      }
```

HCL allows trailing commas in lists. It's preferred to add them so later changes are easier.
```suggestion
      },
```